IPython/core/interactiveshell.py

```diff
@@ -2781,6 +2787,15 @@ def run_ast_nodes(self, nodelist, cell_name, interactivity='last_expr',
         if not nodelist:
             return
 
+        if interactivity == 'last_expr_or_assign':
+            if isinstance(nodelist[-1], ast.Assign):
```

Comment:
I think this is fine for getting this feature in and letting people experiment with it, but be aware that it won't catch `a += 1` (which is an `AugAssign` node), or `a : int = 1` (new in Python 3.6, `AnnAssign` node).